### PR TITLE
Feature/proper email handling

### PR DESCRIPTION
**Pull request by herrLierb**

### CODE REVIEW

File: src/test/java/com/bmw/remotecollab/admin/rest/SessionControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        roomRequest.getEmails().add("INVALID");
+        mockMvc.perform(
+                post(URL_PREFIX + "rooms")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(gson.toJson(roomRequest)))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void testInviteUsersValidation() throws Exception {
```
Comment (mithie) [inline src/test/java/com/bmw/remotecollab/admin/rest/SessionControllerTest.java]: this test fails locally with exception:

[ERROR] testInviteUsersValidation(com.bmw.remotecollab.admin.rest.SessionControllerTest)  Time elapsed: 0.269 s  <<< ERROR!
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is org.thymeleaf.exceptions.TemplateInputException: Error resolving template [d8eb0fa9-ba86-43ad-b647-f1fb292230f9], template might not exist or might not be accessible by any of the configured Template Resolvers
        at com.bmw.remotecollab.admin.rest.SessionControllerTest.testInviteUsersValidation(SessionControllerTest.java:105)
Caused by: org.thymeleaf.exceptions.TemplateInputException: Error resolving template [d8eb0fa9-ba86-43ad-b647-f1fb292230f9], template might not exist or might not be accessible by any of the configured Template Resolvers
        at com.bmw.remotecollab.admin.rest.SessionControllerTest.testInviteUsersValidation(SessionControllerTest.java:105)

File: src/test/java/com/bmw/remotecollab/admin/rest/SessionControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        this.mockMvc.perform(get(URL_PREFIX + "status")).andExpect(status().isOk())
+                .andExpect(content().string(containsString("up")));
+    }
+
+    @Test
+    public void testNewRoomEmailValidation() throws Exception {
+        RequestNewRoom roomRequest = new RequestNewRoom();
+        roomRequest.setRoomName("test123");
+
+        mockMvc.perform(
```
Comment (mithie) [inline src/test/java/com/bmw/remotecollab/admin/rest/SessionControllerTest.java]: this test fails with exception:

[ERROR] Tests run: 4, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 4.591 s <<< FAILURE! - in com.bmw.remotecollab.admin.rest.SessionControllerTest
[ERROR] testNewRoomEmailValidation(com.bmw.remotecollab.admin.rest.SessionControllerTest)  Time elapsed: 0.119 s  <<< ERROR!
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is java.lang.NullPointerException
        at com.bmw.remotecollab.admin.rest.SessionControllerTest.testNewRoomEmailValidation(SessionControllerTest.java:61)
Caused by: java.lang.NullPointerException
        at com.bmw.remotecollab.admin.rest.SessionControllerTest.testNewRoomEmailValidation(SessionControllerTest.java:61)

File: src/main/java/com/bmw/remotecollab/admin/rest/errorhandling/ApiExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ControllerAdvice(basePackageClasses = SessionController.class)
```
[the remark this answers is not in the document]
Comment (herrLierb) [reply src/main/java/com/bmw/remotecollab/admin/rest/errorhandling/ApiExceptionHandler.java]: I was not sure if there are others involved (spring boot side) so I just wanted to not interfere with anything else.. But global would be fine too, if you say so :)

File: src/test/java/com/bmw/remotecollab/admin/rest/SessionControllerTest.java (hunk 1 above)
Comment (herrLierb) [reply src/test/java/com/bmw/remotecollab/admin/rest/SessionControllerTest.java]: fixed it!

File: src/test/java/com/bmw/remotecollab/admin/rest/SessionControllerTest.java (hunk 2 above)
Comment (herrLierb) [reply src/test/java/com/bmw/remotecollab/admin/rest/SessionControllerTest.java]: fixed it!

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -18,4 +18,4 @@ openvidu.secret: MY_SECRET
 ## Email Sending
 email.service.from : noreply.viper-aws-ses@list.bmw.com
 email.service.from.name: Viper Remote Collab
-email.service.link.url: https://localhost:8080/#/session/
\ No newline at end of file
+email.service.link.url: https://ui-standalone.service.viper.bmw.cloud/#/session/
```
[the remark this answers is not in the document]
Comment (herrLierb) [reply src/main/resources/application.properties]: I would like to leave that for now and get it properly injected maybe also via env vars?

File: src/main/java/com/bmw/remotecollab/admin/service/email/EmailList.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import javax.validation.Constraint;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD, ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = EmailListValidator.class)
+public @interface EmailList {
```
[the remark this answers is not in the document]
Comment (herrLierb) [reply src/main/java/com/bmw/remotecollab/admin/service/email/EmailList.java]: added documentation

File: src/main/java/com/bmw/remotecollab/admin/service/email/EmailListValidator.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package com.bmw.remotecollab.admin.service.email;
+
+import org.apache.commons.validator.routines.EmailValidator;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.List;
+
+public class EmailListValidator implements ConstraintValidator<EmailList, List<String>> {
```
[the remark this answers is not in the document]
Comment (herrLierb) [reply src/main/java/com/bmw/remotecollab/admin/service/email/EmailListValidator.java]: added documentation

**Pull request merged**